examples/i2c_HMC5883L.c

```diff
@@ -1,136 +0,0 @@
-/*
- * Author: Brendan Le Foll <brendan.le.foll@intel.com>
- * Copyright (c) 2014 Intel Corporation.
- *
- * Permission is hereby granted, free of charge, to any person obtaining
- * a copy of this software and associated documentation files (the
- * "Software"), to deal in the Software without restriction, including
- * without limitation the rights to use, copy, modify, merge, publish,
- * distribute, sublicense, and/or sell copies of the Software, and to
- * permit persons to whom the Software is furnished to do so, subject to
- * the following conditions:
- *
- * The above copyright notice and this permission notice shall be
- * included in all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
- * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
- * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
- * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
- * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
- */
-
-#include "mraa.h"
-#include "math.h"
-
-#define MAX_BUFFER_LENGTH 6
-#define HMC5883L_I2C_ADDR 0x1E
```

Comment:
TBH I don't have the device to test and since I'm doing the cleanup of examples, I wanna make sure all examples are tested and didn't want to nag @arfoll to test this once again after modification.

If this doesn't makes sense to you people, I am happy to drop mine and add this one as `i2c` based example.

Reply:
I'd suggest to have both, actually, the more - the merrier. @arfoll or @Propanu, do you have access to anything HMC5883L-based to do a checkout (I undofrtunately don't)?

Reply:
Haven't had a hmc5883 sensor for a while (well probably somewher :)) - these days I use the grove RGB lcd thing since I know the i2c codes off by heart. Let's leave 2 for now, we can always remove.